course/models/models.py

```diff
@@ -342,12 +342,12 @@ def save(self, *args, **kwargs):
         if not self.finalized:
             self.calculate_grade(commit=False)
 
-        if not self.in_progress and self.is_correct or self.is_partially_correct:
+        if (not self.in_progress and self.is_correct or self.is_partially_correct) or self.question.event.is_exam:
```

Comment:
You should be really careful about using question.event it might be None. All the questions without an event will face an error here.
also the brackets are wrong, it should be (not self.in_progress) and (self.is_correct or self.is_partially_correct or self.question.event.is_exam)
You don't want to change the grade if its still in progress.